cfme/tests/test_replication.py

```diff
@@ -56,7 +62,44 @@ def test_replication_powertoggle():
             3. VM state changes to on in the Remote and Global appliance.
 .
     """
-    pass
+    instance_name = fauxfactory.gen_alphanumeric(start="test_replication_", length=25).lower()
+    remote_app, global_app = setup_replication
+
+    provider_app_crud(OpenStackProvider, remote_app).setup()
+    provider.appliance = remote_app
+
+    remote_instance = remote_app.collections.cloud_instances.instantiate(
+        instance_name, provider, small_template.name
+    )
+    global_instance = global_app.collections.cloud_instances.instantiate(instance_name, provider)
+
+    # Create instance
+    remote_instance.create_on_provider(find_in_cfme=True)
+    request.addfinalizer(remote_instance.cleanup_on_provider)
+
+    remote_instance.wait_for_instance_state_change(desired_state=remote_instance.STATE_ON)
+
+    # Power OFF instance using global appliance
+    global_instance.power_control_from_cfme(option=global_instance.STOP)
+
+    # Assert instance power off state from both remote and global appliance
+    assert global_instance.wait_for_instance_state_change(
+        desired_state=global_instance.STATE_OFF
+    ).out
+    assert remote_instance.wait_for_instance_state_change(
+        desired_state=remote_instance.STATE_OFF
+    ).out
```

Comment:
Hey @mnadeem92 @valaparthvi 
**Question:**
Do we need assert here with ```.out```?
Aren't we getting timedouterror from ```wait_for_instance_state_change```?
Or is that because of **handle_exception=True** in **wait_for**?

Reply:
`handle_exception=True` does not have to do much here, it will come into action when an exception is encountered during the wait_for by the `func()`, in this case, if any exception is encountered by `_looking_for_state_change`, `handle_exception=True` will handle it. I think it's okay to use `.out` here, I have checked other places and I haven't seen anyone using a try and except block for `wait_for_instance_state_change`.